### PR TITLE
[DECO-1113] Add an init script for jupyter notebooks

### PR DESCRIPTION
**Pull request by kartikgupta-db**
## Changes
* Set's env var IPYTHONDIR to .databricks folder. This allows us to control the jupyter profile and inject our init script, without making it global for the user. 
* Add a small init script that handles some globals. 
* Show prompts to install databricks SDK
* Add a feature manager (depends on dbconnect and databricks sdk) and feature flag.

### CODE REVIEW

File: packages/databricks-vscode/package.json [hunk 1]
```diff
@@ -609,11 +609,13 @@
                         "items": {
                             "enum": [
                                 "debugging.dbconnect",
+                                "notebooks.dbconnect",
                                 "views.cluster",
                                 "views.workspace"
                             ],
                             "enumDescriptions": [
                                 "Interactive debugging using dbconnect V2.",
+                                "Limited local notebook support using dbconnect V2.",
```
Comment (fjakobs) [inline packages/databricks-vscode/package.json:618]:
```suggestion
                                "Limited local notebook support using DB Connect v2.",
```

File: packages/databricks-vscode/package.json [hunk 2]
```diff
@@ -609,11 +609,13 @@
                         "items": {
                             "enum": [
                                 "debugging.dbconnect",
+                                "notebooks.dbconnect",
                                 "views.cluster",
                                 "views.workspace"
                             ],
                             "enumDescriptions": [
                                 "Interactive debugging using dbconnect V2.",
```
Comment (fjakobs) [inline packages/databricks-vscode/package.json:617]:
```suggestion
                                "Interactive debugging using DB Connect v2.",
```

File: packages/databricks-vscode/resources/python/00-databricks-init.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from databricks.sdk.runtime import dbutils
+from databricks.connect import DatabricksSession
+from pyspark.sql import SparkSession, functions as udf
+
+# "display", "displayHTML", "dbutils", "table", "sql", "udf", "getArgument", "sc", "sqlContext", "spark"
```
Comment (fjakobs) [inline packages/databricks-vscode/resources/python/00-databricks-init.py:5]: are these still missing?

Comment (kartikgupta-db) [reply packages/databricks-vscode/resources/python/00-databricks-init.py]: "displayHTML", "table", "sc", "sqlContext" are still missing. Updated the comment.

**Comment by kartikgupta-db** [comment]
> I also added this snippet for proper table rendering:
> 
> ```python
> "from pyspark.sql.connect.dataframe import DataFrame
> 
> def df_html(df):
>   return df.toPandas().to_html()
> 
> html_formatter = get_ipython().display_formatter.formatters['text/html']
> html_formatter.for_type(DataFrame, df_html)
> ```

This is there in a followup PR.

File: packages/databricks-vscode/scripts/writeJupyterInitFileWithVersion.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import {jupyterInitScriptVersion} from "../package.json";
+import path from "node:path";
+
+const pythonDir = path.resolve(__dirname, "..", "resources", "python");
+cpSync(
+    path.join(pythonDir, "00-databricks-init.py"),
+    path.join(
+        pythonDir,
+        "generated",
+        `00-databricks-init-${jupyterInitScriptVersion}.py`
```
Comment (fjakobs) [inline packages/databricks-vscode/scripts/writeJupyterInitFileWithVersion.ts:11]: let's use the hash of the file content instead. That way we don't need to manually maintain a version number and also can easily change the script during development.

File: packages/databricks-vscode/src/extension.ts [hunk 5]
```diff
@@ -231,13 +233,31 @@ export async function activate(
...
+        connectionManager
+    );
+    notebookInitScriptManager.updateInitScript().catch((e) => {
+        NamedLogger.getOrCreate(Loggers.Extension).error(
+            "Failed to update init script",
+            e
+        );
+        if (e instanceof Error) {
+            window.showWarningMessage(
+                `Failed to update databricks notebook init script.` +
```
Comment (fjakobs) [inline packages/databricks-vscode/src/extension.ts:248]:
```suggestion
                `Failed to update databricks notebook init script. ` +
```

File: packages/databricks-vscode/src/language/notebooks/NotebookAccessVerifier.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+              } -> ${await this.pythonExtension.getLatestPackageVersion(
+                  "databricks-sdk"
+              )})`
+            : "";
+
+        const message = `${mainMessagePart} ${envMessagePart}? ${sdkUpdateMessagePart}`;
+        const result = await window.showInformationMessage(
+            message,
+            "Install",
+            "Change environment",
```
Comment (fjakobs) [inline packages/databricks-vscode/src/language/notebooks/NotebookAccessVerifier.ts:52]: After I clicked change environment I was not asked for installing the SDK again. I had to re-open the notebook.

File: packages/databricks-vscode/src/utils/packageJsonUtils.test.ts [hunk 7]
```diff
@@ -53,6 +53,7 @@ describe(__filename, () => {
                 cliArch: "cliArch",
                 vsixArch: "vsixArch",
             },
+            jupyterInitScriptVersion: "0.0.1",
```
Comment (fjakobs) [inline packages/databricks-vscode/src/utils/packageJsonUtils.test.ts:56]: see above

**Review by fjakobs — CHANGES_REQUESTED** [review]

**Pull request merged**